promql/engine.go

```diff
@@ -636,8 +690,21 @@ func (ng *Engine) populateSeries(querier storage.Querier, s *parser.EvalStmt) {
 			// The subqueryOffsetRange function gives the sum of range and the
 			// sum of offset.
 			// TODO(bwplotka): Add support for better hints when subquerying. See: https://github.com/prometheus/prometheus/issues/7630.
-			subqOffset, subqRange := ng.subqueryOffsetRange(path)
+			subqOffset, subqRange, subqMint, subqMaxt := ng.subqueryOffsetRangeTimestamp(path)
 			offsetMilliseconds := durationMilliseconds(subqOffset)
+			if subqMint != math.MaxInt64 && subqMint < hints.Start {
```

Comment:
Why do you need the comparison with MaxInt64? If it is that value, the 2nd clause will never match so it seems redundant.